h2o-genmodel/src/main/java/hex/genmodel/algos/glrm/GlrmLoss.java

```diff
@@ -164,6 +164,18 @@
       sum += Math.max(1 - u[a], 0) - Math.max(1 + u[a], 0);
       return sum;
     }
+    // this function performs the same function as the one above but it is memory optimized for the original
+    // GLRM.java code.  See GLRM.java for details
+    @Override public double mloss(double[] u, int a, int u_len) {
+      if (!(a >= 0 && a < u_len))
+        throw new IndexOutOfBoundsException("a must be between 0 and " + (u_len - 1));
+      double sum = 0;
+//      for (double ui : u)
+      for (int ind=0; ind < u_len; ind++)
+        sum += Math.max(1 + u[ind], 0);
+      sum += Math.max(1 - u[a], 0) - Math.max(1 + u[a], 0);
```

Comment:
It seems like you are multiplying Math.max(1 - u[a], 0) - Math.max(1 + u[a], 0) by u_len and adding it to sum. Could be done outside of this loop, eh.